src/Query/Builder.php

```diff
@@ -1205,10 +1238,18 @@ protected function compileWhereMonth(array $where): array
     {
         extract($where);
 
-        $where['operator'] = $operator;
-        $where['value'] = $value;
+        $value = (int) ltrim($value, '0');
```

Comment:
Is `ltrim()` necessary to avoid interpreting a leading zero as octal notation? I would only expect that to apply to [integer literals](https://www.php.net/manual/en/language.types.integer.php#language.types.integer.syntax).

In my local testing with a PHP 8.2 REPL, the integer cast seems to ignore leading zeroes.